packages/playwright-core/src/server/android/android.ts

```diff
@@ -336,42 +335,58 @@ export class AndroidDevice extends SdkObject {
   }
 
   private async _refreshWebViews() {
+    // possible socketName, eg: webview_devtools_remote_32327, webview_devtools_remote_32327_zeus, webview_devtools_remote_zeus
     const sockets = (await this._backend.runCommand(`shell:cat /proc/net/unix | grep webview_devtools_remote`)).toString().split('\n');
     if (this._isClosed)
       return;
 
     const newPids = new Set<number>();
+    const socketNames = new Set<string>();
     for (const line of sockets) {
-      const match = line.match(/[^@]+@webview_devtools_remote_(\d+)/);
-      if (!match)
-        continue;
-      const pid = +match[1];
-      newPids.add(pid);
-    }
-    for (const pid of newPids) {
-      if (this._webViews.has(pid))
+      const match = line.match(/[^@]+@.*?webview_devtools_remote_?(\d*)/);
+      const matchSocketName = line.match(/[^@]+@(.*?webview_devtools_remote_?.*)/);
+
+      if (!matchSocketName)
         continue;
 
-      const procs = (await this._backend.runCommand(`shell:ps -A | grep ${pid}`)).toString().split('\n');
-      if (this._isClosed)
-        return;
-      let pkg = '';
-      for (const proc of procs) {
-        const match = proc.match(/[^\s]+\s+(\d+).*$/);
-        if (!match)
+      const socketName = matchSocketName[1];
+      socketNames.add(socketName);
+      // possible line: 0000000000000000: 00000002 00000000 00010000 0001 01 5841881 @webview_devtools_remote_zeus
+      // the result: match[1] = ''
+      if (!match || !match[1]) {
+        if (this._webViews.has(socketName))
           continue;
-        const p = match[1];
-        if (+p !== pid)
+        const webView = { pid: -1, pkg: '', socketName };
+        this._webViews.set(socketName, webView);
+        this.emit(AndroidDevice.Events.WebViewAdded, webView);
+        continue;
+      }
+
+      const pid = +match[1];
+      if (!newPids.has(pid)) {
```

Comment:
Now that your map is sourceName => webView the `newPids` map is no longer relevant. Everything should operate sockets instead. So I would:
- extract a function that computes pkg from pid and move it below in this file
- list all sockets, manage Added/Removed events based on sockets. you'll get `newSockets` map.
- optionally resolve pkgs for those that end with `_\d+` so that user could fitler by pkg.

should be less code and much easier to follow.

Reply:
I removed newPids map. Now everything would operate sockets.  Please code review.